common/src/test/java/io/netty/util/HashedWheelTimerTest.java

```diff
@@ -213,6 +213,24 @@ public void testNewTimeoutShouldStopThrowingRejectedExecutionExceptionWhenExisti
         timer.stop();
     }
 
+    @Test()
+    public void reportPendingTimeouts() throws InterruptedException {
+        final HashedWheelTimer timer = new HashedWheelTimer();
+        final Timeout t1 = timer.newTimeout(createNoOpTimerTask(), 100, TimeUnit.MINUTES);
+        final Timeout t2 = timer.newTimeout(createNoOpTimerTask(), 100, TimeUnit.MINUTES);
+
+        assertEquals(2, timer.pendingTimeouts());
+        t1.cancel();
+        Thread.sleep(200);
+
+        assertEquals(1, timer.pendingTimeouts());
+        t2.cancel();
+        Thread.sleep(200);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2925)